presto-main/src/main/java/com/facebook/presto/operator/aggregation/CentralMomentsAggregation.java

```diff
@@ -0,0 +1,88 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation;
+
+import com.facebook.presto.operator.aggregation.state.CentralMomentsState;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.function.AggregationFunction;
+import com.facebook.presto.spi.function.AggregationState;
+import com.facebook.presto.spi.function.CombineFunction;
+import com.facebook.presto.spi.function.Description;
+import com.facebook.presto.spi.function.InputFunction;
+import com.facebook.presto.spi.function.OutputFunction;
+import com.facebook.presto.spi.function.SqlType;
+import com.facebook.presto.spi.type.StandardTypes;
+
+import static com.facebook.presto.operator.aggregation.AggregationUtils.mergeCentralMomentsState;
+import static com.facebook.presto.operator.aggregation.AggregationUtils.updateCentralMomentsState;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+
+@AggregationFunction
+@Description("Returns the central moments of the argument as an array")
+public final class CentralMomentsAggregation
+{
+    private CentralMomentsAggregation() {}
+
+    @InputFunction
+    public static void doubleInput(@AggregationState CentralMomentsState state, @SqlType(StandardTypes.DOUBLE) double value)
+    {
+        updateCentralMomentsState(state, value);
+    }
+
+    @InputFunction
+    public static void bigintInput(@AggregationState CentralMomentsState state, @SqlType(StandardTypes.BIGINT) long value)
+    {
+        updateCentralMomentsState(state, (double) value);
+    }
+
+    @CombineFunction
+    public static void combine(@AggregationState CentralMomentsState state, @AggregationState CentralMomentsState otherState)
+    {
+        mergeCentralMomentsState(state, otherState);
+    }
+
+    @AggregationFunction(value = "skewness")
+    @Description("Returns the skewness of the argument")
+    @OutputFunction(StandardTypes.DOUBLE)
+    public static void skewness(@AggregationState CentralMomentsState state, BlockBuilder out)
+    {
+        long n = state.getCount();
+
+        if (n < 3) {
+            out.appendNull();
+        }
+        else {
+            double result = Math.sqrt(n) * state.getM3() / Math.pow(state.getM2(), 1.5);
+            DOUBLE.writeDouble(out, result);
+        }
+    }
+
+    @AggregationFunction(value = "kurtosis")
+    @Description("Returns the (excess) kurtosis of the argument")
+    @OutputFunction(StandardTypes.DOUBLE)
+    public static void kurtosis(@AggregationState CentralMomentsState state, BlockBuilder out)
+    {
+        double n = state.getCount();
+
+        if (n < 4) {
+            out.appendNull();
+        }
+        else {
+            double m2 = state.getM2();
+            double m4 = state.getM4();
+            double result = (n + 1) * n / (n - 2) * (n - 1) / (n - 3) * m4 / (m2 * m2) - 3 * (n - 1) / (n - 2) * (n - 1) / (n - 3);
```

Comment:
It might be helpful to use some extra parentheses to make the groupings of terms more obvious here. It's not clear to me that the `(n - 1)` term in in the right place.